### PR TITLE
Fix test for br solve for pose, PartialPriorYawPose2 test, and cleanup Pose2Point2Bearing

### LINKED ISSUE #620: Test "test unimodal bearing range factor, solve for pose..." looks wrong
On `BearingRange` from point to pose and fixed point, the pose is free to rotate around the point.

### PR DESCRIPTION
**Pull request by Affie**
close #620

### CODE REVIEW

**Review by dehann — APPROVED** [review]

File: test/testBearingRange2D.jl [hunk 1]
```diff
@@ -314,12 +314,12 @@ end
...
 
 # Start with an empty graph
-N = 1
+N = 75
 fg = initfg()
 
 # Add landmark
 addVariable!(fg, :l1, Point2, tags=[:LANDMARK;])
-addFactor!(fg, [:l1], PriorPoint2(MvNormal([20.0;0.0], Matrix(Diagonal([1.0;1.0].^2)))),  graphinit=false ) # could be IIF.Prior
+addFactor!(fg, [:l1], PriorPoint2(MvNormal([20.0;0.0], Matrix(Diagonal([0.1;0.1].^2)))),  graphinit=false ) # could be IIF.Prior
```
Comment (dehann) [inline test/testBearingRange2D.jl:322]: landmark  at (0,0)

File: test/testBearingRange2D.jl [hunk 2]
```diff
@@ -329,25 +329,32 @@ addVariable!(fg, :x0, Pose2)
...
+p2br = Pose2Point2BearingRange(Normal(0,0.1),Normal(20.0,0.1))
 addFactor!(fg, [:x0; :l1], p2br, graphinit=false)
 
 # there should be just one (the bearingrange) factor connected to :l1
 @test length(ls(fg, :x0)) == 1
 # writeGraphPdf(fg)
 
 # check the forward convolution is working properly
-_pts, = predictbelief(fg, :x0, ls(fg, :x0), N=75)
+_pts, = predictbelief(fg, :x0, ls(fg, :x0); N)
```
Comment (dehann) [inline test/testBearingRange2D.jl:340]: pose should be anywhere on donut ring around the landmark, including at xytheta=(0,0,0)

File: test/testBearingRange2D.jl [hunk 3]
```diff
@@ -329,25 +329,32 @@ addVariable!(fg, :x0, Pose2)
...
-@show abs.(Statistics.mean(pts,dims=2))
-@test sum(abs.(Statistics.mean(pts,dims=2)) .< [2.0; 2.0; 2.0]) == 3
-@show Statistics.std(pts,dims=2)
-@test sum([0.1; 2.0; 0.01] .< Statistics.std(pts,dims=2) .< [5.0; 10.0; 2.0]) == 3
+dists = norm.(eachcol(pts[1:2, :] .- [20,0]))
+@test sum(isapprox.(dists, 20, atol=3)) > N*0.9
+
+# check likelihood at 0,0,0
+#FIXME don't know how this works
+@test_broken getBelief(fg, :x0)([0.0;0.0;0.0;;])[1] < 0.03
```
[the remark this answers is not in the document]
Comment (dehann) [reply test/testBearingRange2D.jl:351]:
```suggestion
@test 0.001 < getBelief(fg, :x0)([0.0;0.0;0.0;;])[1]
```
there should at least be a little bit of probability mass at (0,0,0) one of the infinitely many valid solutions from the convolution

File: test/testBearingRange2D.jl (hunk 1 above)
Comment (Affie) [reply test/testBearingRange2D.jl:322]: Jip, I saw the change was not needed as it's only a single particle on l1. 
I just needed a tighter covariance to get enough density to test for reliably.

File: test/testBearingRange2D.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (Affie) [reply test/testBearingRange2D.jl:351]: Its broken because it returns a NaN

File: test/testBearingRange2D.jl (hunk 2 above)
Comment (Affie) [reply test/testBearingRange2D.jl:340]: Jip. That’s why i had to change the test. It only tested close to 0,0

File: test/testBearingRange2D.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (dehann) [reply test/testBearingRange2D.jl:351]: oh that's unexpected.  It's just a function eval of the kde.  f(x) = Sum_i k(x,xi) where k is Gaussian

**Comment by Affie** [comment]
I'm going ahead and merging this as it's an improvement and the failures are intermittent with at least on this run passing. We can look at the rest as separate PRs

**Pull request merged**